metaworld/envs/mujoco/sawyer_xyz/sawyer_peg_insertion_side_v2.py

```diff
@@ -0,0 +1,241 @@
+import numpy as np
+from gym.spaces import Box
+
+from metaworld.envs.env_util import get_asset_full_path
+from metaworld.envs.mujoco.sawyer_xyz.base import SawyerXYZEnv
+
+
+class SawyerPegInsertionSideEnvV2(SawyerXYZEnv):
+    """
+    Motivation for V2:
+        V1 was completely unsolvable because the observation didn't say where
+        to insert the peg (the hole's location). Furthermore, the hole object
+        could be initialized in such a way that it severely restrained the
+        sawyer's movement.
+    Changelog from V1 to V2:
+        - (6/16/20) Added a 3 element vector to the observation. This vector
```

Comment:
can you use simple L2 distance instead?

Reply:
i.e. 1-element vector

Reply:
Unfortunately not. This is the same issue we had with reach, push, and pick-place. L2 distance is sufficient if we are stateful (guess and check to find correct direction), but in a stateless policy we need the full 3 element vector

Reply:
I suppose we could discretize the 3D space and give each voxel a numeric ID, then pass the ID of the hole's voxel to the observation.

Reply:
okay -- we can revisit these later.

Reply:
Turns out I was wrong. With this env, it doesn't matter whether we overshoot the hole in its axial direction, because the peg will be inserted either way. That eliminates X. And Z is eliminated by the fact that the hole is at a constant height (at least in the V2 env). That leaves only Y, which we can pass as a single number. I've updated the code to reflect this.